aptos-move/aptos-release-builder/src/components/mod.rs

```diff
@@ -3,48 +3,68 @@
 
 use crate::components::feature_flags::Features;
 use anyhow::{anyhow, Result};
-use aptos_types::on_chain_config::{GasScheduleV2, Version};
-use std::path::Path;
+use aptos_types::on_chain_config::{GasScheduleV2, OnChainConsensusConfig, Version};
+use serde::{Deserialize, Serialize};
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::Path,
+};
 
+pub mod consensus_config;
 pub mod feature_flags;
 pub mod framework;
 pub mod gas;
 pub mod version;
 
+#[derive(Serialize, Deserialize, Clone, Eq, PartialEq)]
 pub struct ReleaseConfig {
+    pub testnet: bool,
+    pub framework_release: bool,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub gas_schedule: Option<GasScheduleV2>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub version: Option<Version>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
     pub feature_flags: Option<Features>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub consensus_config: Option<OnChainConsensusConfig>,
 }
 
 impl ReleaseConfig {
-    pub fn generate_release_proposal_scripts(
-        &self,
-        base_path: &Path,
-        is_testnet: bool,
-    ) -> Result<()> {
+    pub fn generate_release_proposal_scripts(&self, base_path: &Path) -> Result<()> {
         let mut result = vec![];
```

Comment:
you are not returning the result, or doing anything else with it?

Reply:
This function will generate a set of move scripts into the `base_path`.